### ISSUE TITLE
Question about HyenaDNA working

### ISSUE THREAD

**Issue by varuntrivedi18** [HazyResearch/hyena-dna#28]
Hello,

I have been using HyenaDNA to pre-train it on my custom dataset, and I wanted some insight into its working based on the output logs from the pre-training run. For every epoch, after updating model weights based on the training data, the tool evaluates the performance on the validation data by calculating `val/loss`. In the process, it loads validation data from two different dataloaders (`Validation Dataloader 0` and `Validation Dataloader 1`). I provided the test data separately from the validation data, and after varying the train : validation : test splits, I found that the size of `Validation Dataloader 1` changes in proportion to the size of the test set. Why does this happen, given that the test set is separate from the validation set, and that the tool reports a separate value of `test/loss` after every epoch?

**Comment by exnx** [comment]
I don't really understand your issue.

`Validation Dataloader 1` is the test dataloader, that's just how Lightning handles it.

Are you using the Hg38 or species dataloader?

For HG38, take a look [here](https://github.com/HazyResearch/hyena-dna/blob/eed1f4b357706bf5f5e5cb057d6ef52ec42acdd9/src/dataloaders/genomics.py#L127).  It's similar for the species dataloader.

This defines exactly what dataset is used for each dataloader.  Make sure you're passing the right dataset object to the right dataloader.  This is the same for Pytorch dataloaders too, vs Lightning.

I recommend adding breakpoints and stepping through the code line by line in the dataloader to see the objects are exactly as you expect.

**Comment by varuntrivedi18** [comment]
Thank you for the clarification! I'm new to deep learning and wasn't aware that `Validation Dataloader 1` indeed is the test dataloader, not the validation dataloader.

**Comment by varuntrivedi18** [comment]
Thank you for the clarification! I'm new to deep learning and wasn't aware that `Validation Dataloader 1` indeed is the test dataloader, not the validation dataloader.
